Assets/MRTK/Tests/PlayModeTests/ScrollViewTests.cs

```diff
@@ -770,6 +770,232 @@ public IEnumerator ScrollEngageOnlyFromFrontInteraction()
+
+            GameObject button1 = InstantiatePrefab(TestButtonUtilities.PressableHoloLens2PrefabPath);
+            button1.transform.parent = scrollObject.transform;
+
+            GameObject button2 = InstantiatePrefab(TestButtonUtilities.PressableHoloLens2PrefabPath);
+            button2.transform.parent = scrollObject.transform;
+
+            // Setting up a vertical 1x1 scroll view. 
+            var scrollView = scrollObject.AddComponent<ScrollingObjectCollection>();
+            scrollView.ScrollDirection = ScrollingObjectCollection.ScrollDirectionType.UpAndDown;
+            scrollView.CellWidth = button1.GetComponent<NearInteractionTouchable>().Bounds.x;
+            scrollView.CellHeight = button1.GetComponent<NearInteractionTouchable>().Bounds.y;
+            scrollView.ItemsPerTier = 1;
+            scrollView.ViewableArea = 1;
+            scrollView.UpdateCollection();
+            scrollObject.transform.position = - Vector3.up;
+            scrollObject.transform.rotation = Quaternion.LookRotation(-Vector3.up);
+
+            // Setting up camera to look down
+            MixedRealityPlayspace.Position = Vector3.zero;
+            MixedRealityPlayspace.Rotation = Quaternion.LookRotation(-Vector3.up);
+
+            PressableButton button1Component = button1.GetComponentInChildren<PressableButton>();
+
+            Assert.IsNotNull(button1Component);
+
+            // Hand positions
+            float offset = 0.001f;
+            Vector3 initialPos = Vector3.zero;
+            Vector3 preButtonPressPos = button1Component.transform.position - new Vector3(0, button1Component.StartPushDistance - offset, 0);
+            Vector3 pastButtonPressPos = button1Component.transform.position - new Vector3(0, button1Component.PressDistance + 0.03f, 0);
+            Vector3 scrollEngagedPos = pastButtonPressPos + Vector3.forward * (scrollView.HandDeltaMagThreshold + scrollView.CellHeight + offset);
+
+            // Moving hand along z axis should still engage an up-down scroll view rotated 90 degrees around x
+            TestHand hand = new TestHand(Handedness.Right);
+            yield return hand.Show(initialPos);
+            yield return hand.MoveTo(preButtonPressPos);
+            yield return hand.MoveTo(pastButtonPressPos);
+            yield return hand.MoveTo(scrollEngagedPos);
+
+            Assert.IsTrue(scrollView.isDragging, "Scroll view is not being dragged.");
+
+            yield return hand.Hide();
+        }
+
+        /// <summary>
+        /// Tests if scroll can be moved by page, by tier or to make specific element to be presented in the first visible tier.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator CanBeScrolledByTierOrIndexOrPage()
+        {
+            // Setting up nine pressable buttons as scroll items
+            GameObject scrollObject = new GameObject();
+
+            GameObject button1 = InstantiatePrefab(TestButtonUtilities.PressableHoloLens2PrefabPath);
```

Comment:
Consider factoring out common code into a helper method or function here, and below.